src/toolkit/facility_cost.cycpp.h

```diff
@@ -1,19 +1,159 @@
+
+#pragma cyclus var {                                                \
+    "default" : 0.0, "uilabel" : "Annual O&M Cost in dollars",     \
+    "doc" : "Annual O&M Cost required to run facility in dollars", \
+    "units" : "$USD" }
+double operations_and_maintenance;
+
+#pragma cyclus var { \
+    "default": 1.0, \
+    "uilabel": "Estimated Useful lifetime of facility for economic purposes in years", \
+    "doc": "Estimate on how long the facility will be active for economic purposes", \
+    "units": "years" \
+    }
+double facility_lifetime;
+
+#pragma cyclus var { \
+    "default": 1.0, \
+    "uilabel": "Annual fractional increase of initial facility capacity", \
+    "doc": "Fraction of initial capacity by which facility capacity increases/decreases each year", \
+    "units": "Dimensionless" \
+    }
+double capacity_decline_factor;
 
 #pragma cyclus var { \
-"default": 0.0, \
-"uilabel": "Capital cost required to build facility", \
-"doc": "Capital cost required to build facility" \
+    "default": 0.0, \
+    "uilabel": "Cost in dollars of labor required to produce one unit of production", \
+    "doc": "Cost in dollars of labor required to produce one unit of production", \
+    "units": "$USD" \
+    }
+double per_unit_labor_cost;
+
+
+#pragma cyclus var { \
+    "default": 0.04, \
+    "uilabel": "Compounding fractional annual increase in the cost of labor as a decimal", \
+    "doc": "Cost in dollars of labor required to produce one unit of production", \
+    "units": "Dimensionless" \
+    }
+double annual_labor_cost_increase_factor;
+
+
+// Must be done in a function so that we can access the user-defined values
+std::unordered_map<std::string, double> InitializeParamList() const override {
+  std::unordered_map<std::string, double> econ_params{
+      {"capital_cost", capital_cost},
+      {"operations_and_maintenance", operations_and_maintenance},
+      {"facility_lifetime", facility_lifetime},
+      {"capacity_decline_factor", capacity_decline_factor},
+      {"per_unit_labor_cost", per_unit_labor_cost},
+      {"annual_labor_cost_increase_factor", annual_labor_cost_increase_factor}};
+
+  return econ_params;
+}
+
+double ComputeDelta(double depreciation_constant, double initial_book_value,
+                              int T_hat, double r, double alpha) const {
+  std::vector<double> book_value = {initial_book_value};
+  std::vector<double> dt;
+
+  double discount_factor = 1.0 / (1.0 + r);
+
+  for (int i = 0; i < T_hat - 1; ++i) {
+    int current_depreciation = -depreciation_constant * book_value[i];
+    book_value.push_back(book_value[i] + current_depreciation);
+    dt.push_back(current_depreciation / book_value[0]);
+  }
+  
+  double delta_partial_sum = 0.0;
+  for (int t = 1; t < T_hat; ++t) {
+    delta_partial_sum += dt[t - 1] * std::pow(discount_factor, t); 
+  }
+
+  return (1 - alpha * -delta_partial_sum) / (1 - alpha);
+}
+
+double GetCost(double units_of_production, double input_cost) {
```

Comment:
By contrast, we should inject this (and not put it in the `EconomicEntity` class), because it depends on the specific class variables that we are injecting.